packages/preview-service/services/taskManager.ts

```diff
@@ -0,0 +1,81 @@
+import type { UpdateHealthcheckData } from 'clients/execHealthcheck'
+import {
+  metricDuration,
+  metricOperationErrors
+} from '../observability/prometheusMetrics'
+import type { NotifyUpdate, UpdatePreviewMetadata } from '../repositories/objectPreview'
+import type { GetNextUnstartedObjectPreview } from '../repositories/objectPreview'
+import type { GenerateAndStore360Preview } from './360preview'
+import type { Logger } from 'pino'
+
+let shouldExit = false
+
+export function forceExit() {
+  shouldExit = true
+}
+
+type RepeatedlyPollForWork = () => Promise<void>
+export const repeatedlyPollForWorkFactory =
+  (deps: {
+    updateHealthcheckData: UpdateHealthcheckData
+    getNextUnstartedObjectPreview: GetNextUnstartedObjectPreview
+    generateAndStore360Preview: GenerateAndStore360Preview
+    updatePreviewMetadata: UpdatePreviewMetadata
+    notifyUpdate: NotifyUpdate
+    onExit: () => void
+    logger: Logger
+  }): RepeatedlyPollForWork =>
+  async () => {
+    if (shouldExit) {
+      deps.onExit()
+      return
+    }
+
+    try {
+      const task = await deps.getNextUnstartedObjectPreview()
+
+      // notify the healthcheck that we are still alive
+      deps.updateHealthcheckData()
+
+      if (!task) {
+        setTimeout(repeatedlyPollForWorkFactory(deps), 1000)
```

Comment:
weird/smelly logic here, but i assume this is largely a copy of what existed before.

smelly cause:
1. Duplicate `setTimeout(...)` calls that could be extracted to a parameterizable function `await timeout(xx)`
2. Devs shouldn't have to (remember to) manually invoke the next loop of the timeout. In fact, the actual looping/timeout bit feels like a separate concern altogether. I'd probably make this function just return a status enum, and then a different function would handle invoking this one, and setting up the correct setTimeout() depending on the status enum